src/systemTest/java/com/flightstats/hub/model/ChannelItemPathParts.java

```diff
@@ -1,24 +1,92 @@
 package com.flightstats.hub.model;
 
+import com.flightstats.hub.util.TimeUtil;
 import lombok.Builder;
-import lombok.Getter;
+import lombok.Value;
+import okhttp3.HttpUrl;
 import org.joda.time.DateTime;
 
-@Getter
+import java.util.Optional;
+import java.util.function.Function;
+
+@Value
 @Builder
 public class ChannelItemPathParts {
     String path;
-    String timePath;
-
-    String channelName;
-    String hashKey;
-    int year;
-    int month;
-    int day;
-    int hour;
-    int minute;
-    int second;
-    int millis;
-
-    DateTime dateTime;
+
+    public String getChannelName() {
+        return getPath().substring(0, getPath().indexOf("/"));
```

Comment:
This all looks much better (well, to my eye). A more cohesive handling of what a Hub path means in one place.